pulp_container/app/registry_api.py

```diff
@@ -620,35 +622,56 @@ def partial_update(self, request, path, pk=None):
         Process a chunk that will be appended to an existing upload.
         """
         _, repository = self.get_dr_push(request, path)
+        upload = get_object_or_404(models.Upload, repository=repository, pk=pk)
         chunk = request.META["wsgi.input"]
-        if "Content-Range" in request.headers or "digest" not in request.query_params:
-            whole = False
-        else:
-            whole = True
+        if range_header := request.headers.get("Content-Range"):
+            found = self.content_range_pattern.match(range_header)
+            start = int(found.group(1))
+            end = int(found.group(2))
+            length = end - start + 1
 
-        if whole:
-            start = 0
         else:
-            content_range = request.META.get("HTTP_CONTENT_RANGE", "")
-            match = self.content_range_pattern.match(content_range)
-            start = 0 if not match else int(match.group("start"))
-
-        upload = get_object_or_404(models.Upload, repository=repository, pk=pk)
+            # is this header always present?
+            length = int(request.headers["Content-Length"])
+            start = 0
 
-        chunk = ContentFile(chunk.read())
         with transaction.atomic():
             if upload.size != start:
                 raise Exception
 
-            upload.append(chunk, upload.size)
-            upload.size += chunk.size
+            # if more chunks
+            if range_header:
```

Comment:
What if the range header is there, but it still covers all the blob. I think we can do the same trick. Also if an artifact is set here, i think we should reject further chunks.

Reply:
You mean there is a range header but only one chunk will be ever sent? I don't think this would be a valid case

Reply:
If you don't mind I would rather refrain from adding any additional logic for hypothetical cases. This is already not the most straightforward as is but at least there is a practical use of that code-path.